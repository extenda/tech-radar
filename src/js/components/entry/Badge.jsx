import React from 'react';
import PropTypes from 'prop-types';
import Icon from '../Icon';

const Badge = (props) => {
  const {
    className,
    icon,
    text,
    link,
  } = props;

  let content;

  if (link) {
    content = (
      <a href={link}>{text}</a>
    );
  } else {
    content = text;
  }

  return (
    <span className={`badge ${className}`}>
      <Icon name={icon} spaceBefore={false} />
      {content}
    </span>
  );
};

Badge.propTypes = {
  className: PropTypes.string,
  icon: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  link: PropTypes.string,
};

Badge.defaultProps = {
  className: '',
  link: null,
};

export default Badge;
