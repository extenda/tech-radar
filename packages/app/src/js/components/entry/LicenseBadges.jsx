import React from 'react';
import License from './License';
import Badge from './Badge';

const LicenseBadges = (props) => {
  const {
    license: { openSource, commercial },
  } = props;

  return (
    <>
      {openSource && (
        <Badge
          icon="balance-scale"
          className="license"
          text={openSource.name}
          link={openSource.link}
        />
      )}
      {commercial && (
        <Badge icon="balance-scale" className="license" text="Commercial" />
      )}
    </>
  );
};

LicenseBadges.propTypes = {
  ...License.propTypes,
};

LicenseBadges.defaultProps = {
  ...License.defaultProps,
};

export default LicenseBadges;
