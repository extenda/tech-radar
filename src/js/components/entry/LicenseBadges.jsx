import React from 'react';
import License from './License';
import Badge from './Badge';

const LicenseBadges = (props) => {
  const { license: { openSource, commercial } } = props;

  return (
    <React.Fragment>
      {openSource && (
        <Badge icon="balance-scale" className="license" text={openSource.name} link={openSource.link} />
      )}
      {commercial && (
        <Badge icon="balance-scale" className="license" text="Commercial" />
      )}
    </React.Fragment>
  );
};

LicenseBadges.propTypes = {
  ...License.propTypes,
};

LicenseBadges.defaultProps = {
  ...License.defaultProps,
};

export default LicenseBadges;
