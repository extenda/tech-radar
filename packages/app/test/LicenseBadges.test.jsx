import { shallow } from 'enzyme';
import React from 'react';

import Badge from '../src/js/components/entry/Badge';
import LicenseBadges from '../src/js/components/entry/LicenseBadges';

describe('<LicenseBadges />', () => {
  test('It renders no badges for missing license', () => {
    const component = shallow(<LicenseBadges />);
    expect(component).toMatchSnapshot();
    expect(component.find(Badge)).toHaveLength(0);
  });

  test('It renders open-source badge', () => {
    const component = shallow(
      <LicenseBadges
        license={{
          openSource: {
            name: 'MIT',
            link: 'https://license.com',
          },
        }}
      />,
    );

    expect(component).toMatchSnapshot();
    expect(component.find(Badge)).toHaveLength(1);
  });

  test('It renders commercial badge', () => {
    const component = shallow(
      <LicenseBadges
        license={{
          commercial: {
            company: 'Oracle',
          },
        }}
      />,
    );

    expect(component).toMatchSnapshot();
    expect(component.find(Badge)).toHaveLength(1);
  });

  test('It renders dual license', () => {
    const component = shallow(
      <LicenseBadges
        license={{
          commercial: {
            company: 'Oracle',
          },
          openSource: {
            name: 'MIT',
            link: 'https://license.com',
          },
        }}
      />,
    );

    expect(component.find(Badge)).toHaveLength(2);
  });
});
