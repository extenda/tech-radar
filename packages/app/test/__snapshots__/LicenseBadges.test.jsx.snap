// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LicenseBadges /> It renders commercial badge 1`] = `
<Fragment>
  <Badge
    className="license"
    icon="balance-scale"
    link={null}
    text="Commercial"
  />
</Fragment>
`;

exports[`<LicenseBadges /> It renders no badges for missing license 1`] = `<Fragment />`;

exports[`<LicenseBadges /> It renders open-source badge 1`] = `
<Fragment>
  <Badge
    className="license"
    icon="balance-scale"
    link="https://license.com"
    text="MIT"
  />
</Fragment>
`;
